tck/src/main/java/io/a2a/tck/server/AgentExecutorProducer.java

```diff
@@ -42,22 +43,12 @@ public void execute(RequestContext context, EventQueue eventQueue) throws JSONRP
                 eventQueue.enqueueEvent(task);
             }
 
+            TaskUpdater updater = new TaskUpdater(context, eventQueue);
+
             // Immediately set to WORKING state
-            eventQueue.enqueueEvent(new TaskStatusUpdateEvent.Builder()
-                    .taskId(context.getTaskId())
-                    .contextId(context.getContextId())
-                    .status(new TaskStatus(TaskState.WORKING))
-                    .build());
-            
+            updater.startWork();
             System.out.println("====> task set to WORKING, starting background execution");
 
-//            // Fire and forget - start the task but don't wait for it
-//            CompletableFuture<Void> taskFuture = CompletableFuture
-//                .runAsync(() -> executeTaskInBackground(context, eventQueue), taskExecutor);
-            
-//            // Store the future for potential cancellation
-//            runningTasks.put(context.getTaskId(), taskFuture);
-            
             // Method returns immediately - task continues in background
```

Comment:
There's a executeTaskInBackground method in this class but it looks to be unused?